crates/driver/src/domain/competition/mod.rs

```diff
@@ -211,13 +213,20 @@ impl Competition {
             })
             .unzip();
 
-        self.settlement.lock().unwrap().clone_from(&settlement);
-
-        let settlement = match settlement {
-            Some(settlement) => settlement,
+        let Some(settlement) = settlement else {
             // Don't wait for the deadline because we can't produce a solution anyway.
-            None => return Ok(score),
+            return Ok(score);
         };
+        let solution_id = settlement.solution().get();
```

Comment:
This one should be unique, but I still see a lot of duplicates in the logs, like `Solution { id: Single(0) ...`. We have a single instance of `Competition`, right? I think I missed the part where IDs get reset.

Reply:
These Ids can only get reset when a new binary gets started. I checked locally and `cargo nextest` and `cargo test` behave slightly differently. `cargo test` seemingly instantiates the binary only ones and then runs all tests in the same process which results in no ids getting recycled during tests.
Whereas `cargo nextest` runs each test in a separate process which means on every test the Id starts from 0.
Discussed [here](https://github.com/nextest-rs/nextest/issues/709)

Reply:
> which means on every test the Id starts from 0

Yeah, but I am about the [prod logs](https://aws-es.cow.fi/_dashboards/goto/3db587a36299caadb88b9c30cf375029?security_tenant=global). There, I see a lot of duplicates of the same solver and pod.

Reply:
The code was not merged yet so why should the currently incorrect solution ids already be fixed?
BTW this is the [line](https://github.com/cowprotocol/services/pull/3016/files#diff-b28e258782f144c493f504e1d865601f0f557b178a43a255bc38497d0dc48fd6L16) that made all the solutions get the id `0` so far. Since we didn't look up the id anyway it didn't matter which value we return here.

Reply:
> BTW this is the [line](https://github.com/cowprotocol/services/pull/3016/files#diff-b28e258782f144c493f504e1d865601f0f557b178a43a255bc38497d0dc48fd6L16) that made all the solutions get the id 0

Oh, nice, thanks. That is exactly what I missed 😄